src/main/java/meditracker/command/AddCommand.java

```diff
@@ -52,17 +70,89 @@ public AddCommand(String arguments) throws MediTrackerException {
+     * @throws NullPointerException  if any of the required arguments are null.
+     */
+    private Medication createMedication() throws NumberFormatException, NullPointerException {
+        medicationName = parsedArguments.get(ArgumentName.NAME);
+        String expiryDate = parsedArguments.get(ArgumentName.EXPIRATION_DATE);
+        String intakeFreq = parsedArguments.get(ArgumentName.INTAKE_FREQUENCY);
+        String remarks = parsedArguments.get(ArgumentName.REMARKS);
+        String repeat = parsedArguments.get(ArgumentName.REPEAT);
+
+        String medicationQuantityArg = parsedArguments.get(ArgumentName.QUANTITY);
+        String medicationDosageArg = parsedArguments.get(ArgumentName.DOSAGE);
+        String medicationDosageMorningArg = parsedArguments.get(ArgumentName.DOSAGE_MORNING);
+        String medicationDosageAfternoonArg = parsedArguments.get(ArgumentName.DOSAGE_AFTERNOON);
+        String medicationDosageEveningArg = parsedArguments.get(ArgumentName.DOSAGE_EVENING);
+
+        parseStringToValues(medicationQuantityArg, medicationDosageArg, medicationDosageMorningArg,
+                medicationDosageAfternoonArg, medicationDosageEveningArg);
+
+        return new Medication(medicationName, medicationQuantity, medicationDosage,
+                medicationDosageMorning, medicationDosageAfternoon, medicationDosageEvening,
+                expiryDate, intakeFreq, repeat, remarks);
+    }
+
+    /**
+     * Performs assertion tests for medication and daily medication managers.
+     *
+     * @param medicationManager      The MedicationManager object representing the list of medications.
+     * @param dailyMedicationManager The DailyMedicationManager object representing the list of daily medications.
+     */
+    private void assertionTest(MedicationManager medicationManager, DailyMedicationManager dailyMedicationManager) {
         assert medicationManager.getTotalMedications() != 0 : "Total medications in medication " +
                 "manager should not be 0!";
-        Ui.showAddCommandMessage();
+        assert dailyMedicationManager.getTotalDailyMedication() != 0 : "Total medications in daily medication " +
+                "manager should not be 0!";
+    }
+
+    /**
+     * Parses string values to double for medication attributes.
+     *
+     * @param medicationQuantity      The quantity of the medication.
+     * @param medicationDosage        The dosage of the medication.
+     * @param medicationDosageMorning The morning dosage of the medication.
+     * @param medicationDosageAfternoon The afternoon dosage of the medication.
+     * @param medicationDosageEvening The evening dosage of the medication.
+     * @throws NumberFormatException if there is an error in parsing numeric values.
+     * @throws NullPointerException  if any of the required arguments are null.
+     */
+    private void parseStringToValues(String medicationQuantity, String medicationDosage,
```

Comment:
Thx for all the changes! I believe u have changed the var types to Double right, so I think this method is no longer necessary?

Reply:
I believe the method is necessary because when reading in user input using nextLine(), the command is read in as a string. Hence parsing from string to double is still necessary.